src/index.js

```diff
@@ -1,44 +1,56 @@
 const LANGUAGES_LIST = require('./data.js');
 
+const LANGUAGES = {};
+const LANGUAGES_BY_NAME = {};
+const LANGUAGE_CODES = [];
+const LANGUAGE_NAMES = [];
+const LANGUAGE_NATIVE_NAMES = [];
+
+for (const code in LANGUAGES_LIST) {
+  const { name, nativeName } = LANGUAGES_LIST[code];
+  LANGUAGES[code] =
+    LANGUAGES_BY_NAME[name.toLowerCase()] =
+    LANGUAGES_BY_NAME[nativeName.toLowerCase()] =
+      { code, name, nativeName };
+  LANGUAGE_CODES.push(code);
+  LANGUAGE_NAMES.push(name);
+  LANGUAGE_NATIVE_NAMES.push(nativeName);
+}
+
 module.exports = class ISO6391 {
   static getLanguages(codes = []) {
-    return codes.map(code => ({
-      code,
-      name: ISO6391.getName(code),
-      nativeName: ISO6391.getNativeName(code),
-    }));
+    return codes.map(code =>
+      ISO6391.validate(code)
+        ? Object.assign({}, LANGUAGES[code])
+        : { code, name: '', nativeName: '' }
+    );
   }
 
   static getName(code) {
     return ISO6391.validate(code) ? LANGUAGES_LIST[code].name : '';
   }
 
   static getAllNames() {
-    return Object.values(LANGUAGES_LIST).map(l => l.name);
+    return LANGUAGE_NAMES.slice();
   }
 
   static getNativeName(code) {
     return ISO6391.validate(code) ? LANGUAGES_LIST[code].nativeName : '';
   }
 
   static getAllNativeNames() {
-    return Object.values(LANGUAGES_LIST).map(l => l.nativeName);
+    return LANGUAGE_NATIVE_NAMES.slice();
   }
 
   static getCode(name) {
-    const code = Object.keys(LANGUAGES_LIST).find(code => {
-      const language = LANGUAGES_LIST[code];
-
-      return (
-        language.name.toLowerCase() === name.toLowerCase() ||
-        language.nativeName.toLowerCase() === name.toLowerCase()
-      );
-    });
-    return code || '';
+    name = name.toLowerCase();
+    return LANGUAGES_BY_NAME.hasOwnProperty(name)
+      ? LANGUAGES_BY_NAME[name].code
+      : '';
```

Comment:
The original code also supports getting code by native name. So in the new code, should also check `LANGUAGE_NATIVE_NAMES.hasOwnProperty(name)`

Reply:
This case is handled as well. When `LANGUAGES_BY_NAME` is constructed I use the native name as key as well so we handle both cases and save an extra lookup.